integration_test/Dialect/Handshake/sync_op/sync_op.mlir

```diff
@@ -0,0 +1,44 @@
+// REQUIRES: iverilog,cocotb
+
+// This test is executed with all different buffering strategies
+
+// RUN: hlstool %s --ir-input-level=1 --dynamic --verilog --lowering-options=disallowLocalVariables > %t.sv && \
+// RUN: %PYTHON% %S/../cocotb_driver.py --objdir=%T --topLevel=top --pythonModule=sync_op --pythonFolder=%S %t.sv 2>&1 | FileCheck %s
+
+// CHECK: ** TEST
+// CHECK: ** TESTS=[[N:.*]] PASS=[[N]] FAIL=0 SKIP=0
+
+handshake.func @top(%arg0: i64, %arg1: none, ...) -> (i64, none) attributes {argNames = ["in0", "inCtrl"], resNames = ["out0", "outCtrl"]} {
```

Comment:
this is a very large test for just testing the `sync` op - can it be reduced in size? I'd hope for any new tests to avoid going the route of the existing set of `handshake` tests which are close to unmaintainable.

Reply:
If that's the case, then there should probably be two integration tests - one for "does it work" which is probably just a few IR ops (e.g. two syncs and some buffers) and then a separate "reordering".